internal/util/sysreadfile.go

```diff
@@ -35,11 +38,25 @@ func SysReadFile(file string) (string, error) {
 	//
 	// Since we either want to read data or bail immediately, do the simplest
 	// possible read using syscall directly.
-	b := make([]byte, 128)
+	const sysFileBufferSize = 128
+	b := make([]byte, sysFileBufferSize)
 	n, err := syscall.Read(int(f.Fd()), b)
 	if err != nil {
 		return "", err
 	}
 
 	return string(bytes.TrimSpace(b[:n])), nil
 }
+
+// ReadFileNoStat uses ioutil.ReadAll to read contents of entire file.
+// This is similar to ioutil.ReadFile but without the call to os.Stat, because
+// many files in /proc and /sys report incorrect file sizes (either 0 or 4096).
+func ReadFileNoStat(filename string) ([]byte, error) {
+	f, err := os.Open(filename)
+	if err != nil {
+		return nil, err
+	}
+	defer f.Close()
+
+	return ioutil.ReadAll(f)
```

Comment:
It would probably be a good idea to put a limit on this of like 32/64/128 MiB or something via io.LimitReader, just as a sanity check.

Reply:
Seems like a good idea.  I think we could make it even smaller like 500kB?  Any larger and we should be using a scanner instead of reading the whole thing at once.